Kernel/Arch/PageFault.h

```diff
@@ -43,22 +52,48 @@ class PageFault {
         Write = PageFaultFlags::Write,
     };
 
+    enum class Mode {
```

Comment:
This could be the same thing as the "PreviousMode" if "PreviousMode" had a better name :)

Reply:
This code now uses the ExecutionMode enum :)